### ISSUE TITLE
No remote video feed, IVideoFrameObserver::onRenderVideoFrame never called on Windows

### ISSUE THREAD

**Issue by willcowling** [AgoraIO-Community/Agora-Unreal-SDK#16]
Hi,

Can you confirm this demo code is working as expected on Windows in Unreal 4.26?  For me, testing on multiple different machines, there is no remote video feed as the onRenderVideoFrame for the VideoFrameObserver is never called.  Some older sample code I have works ok.  Testing with SDK 3.5.1.  Have set valid APP_ID and am using a Temp Token generated from the Console. The underlying connection works, but just not getting the onRenderVideoFrame callbacks with this sample code.  Local video renders correctly and onCaptureVideoFrame is called fine.

Can you help?

**Comment by alexhajdu** [comment]
Hi, I've had difficulties running the demo as well. I've one client build on my phone and running another app instance in the UE4 editor. On my mobile build, there is picture flickering; switching between computer cam and local (phone) cam picture. On my computer, I can't see the stream coming from remote (phone) at all. 

@WillCowling Can you confirm this flickering on your android build, please?

**Comment by alexhajdu** [comment]
Also, I can confirm, that this is the behavior after the user joins:

```
LogTemp: Warning: UVideoViewUserWidget::onUserJoined ======1493317571
LogTemp: Warning: AgoraController::SetupRemoteVideo  uid: 1493317571
LogTemp: Warning: VideoFrameObserver::addUserInfo uid: 1493317571
```

but `IVideoFrameObserver::onRenderVideoFrame` is never called.

**Comment by alexhajdu** [comment]
VideoFrameObserver::onTick: 
`if (iter->second._texture && iter->second._fresh) {  ... }`  
second._texture is nullptr and second._fresh is false

**Comment by ToposStudio** [comment]
This is bizarre. We noticed this on a test we're doing right now, also with 4.26. Strangely the built package we have on oculus quest is able to receive video, but if you notice the issue I created the video flickers. The desktop client we created and running it in the unreal editor both result in no remote video received. A few weeks ago we tested the blueprint version of this (no longer supported) and that was working, we had a windows client and completed a video chat.

**Comment by alexhajdu** [comment]
Yes, I can confirm the previous version of plugin, built upon BPs was working for us too. We needed to rework from scratch, which is bearable. But this bug is show stopper.

**Comment by ToposStudio** [comment]
On the quest we have no other option that I can find for video chat in unreal as the web browser is apparently broken. if the only thing missing is that call, can we manually set it up somehow?

**Comment by ToposStudio** [comment]
It might just be me, but it seems like this looks somewhat connected

rtcengnine calls the videoframeobserver ontick. 

```
void AgoraRtcEngine::OnTick() {
    if (m_videoFrameObserver_ptr) {
        m_videoFrameObserver_ptr->onTick();
    }
}
```

The videofromeobserverontick calls this: 

```
std::map<unsigned int, VideoTextureFrame>::iterator iter;
        for(iter = _users_texture_frame_map.begin(); iter != _users_texture_frame_map.end(); iter ++) {
            if (iter->first != 0) {
                if (iter->second._texture && iter->second._fresh) {
                    _user_video_texture_observer->OnRemoteVideoTextureUpdate(iter->first, iter->second);
                }
            }
        }
```

This seems to connect back to this: 

`    virtual void OnRemoteVideoTextureUpdate(unsigned int uid, const VideoTextureFrame& _videoFrame){}`

Which is somehow connected to this: 

`void AgoraController::OnRemoteVideoTextureUpdate(unsigned int uid, const agora::rtc::ue4::VideoTextureFrame& _videoFrame)
{
	UE_LOG(LogTemp, Warning, TEXT("OnRemoteVideoTextureUpdate uid: %u"), uid);
	if (_videoWidget) {
		_videoWidget->OnRemoteVideoTextureUpdate(uid, _videoFrame);
	}
}`

Could there be something wrong with the code in the ontick? Is it not iterating properly or something?

**Comment by ToposStudio** [comment]
Okay I just ran through debugging.

I started one copy of my project with the webcam enabled and then started a second copy with debugging to break on this line: 

 if (_user_video_texture_observer) 
 
 I think I'm seeing an issue here. Even with a remote user, who is sending video (or who should be)
 
 the receiving client seems to fail on line 99 

`if (iter->first != 0) {`

even with a connected user supposedly sending video, iter->first always equals 0, so it never gets into doing anything with the remote video frame that i can see. 

if i change that to == 0 then I start getting some calls: OnRemoteVideoTextureUpdate uid: 0 but it comes out like this and I still can't see any video. 

What's truly bizarre is why android/quest shows anything and desktop shows nothing.

**Comment by ToposStudio** [comment]
> Yes, I can confirm the previous version of plugin, built upon BPs was working for us too. We needed to rework from scratch, which is bearable. But this bug is show stopper.

They've asked us for some logs through Agora support, but nothing yet. In the meantime, we've gone back to the blueprint version. It works. It gives us the features we need for now, so we'll stick with that until they fix this. I just tested a windows->quest video chat and the video came through clear and without flickering on the quest. Of course windows to windows works fine.
